nautobot/circuits/views.py

```diff
@@ -4,277 +4,189 @@
+    filterset_form_class = forms.ProviderFilterForm
+    form_class = forms.ProviderForm
+    queryset = Provider.objects.annotate(count_circuits=count_related(Circuit, "provider"))
+    serializer_class = serializers.ProviderSerializer
+    table_class = tables.ProviderTable
 
     def get_extra_context(self, request, instance):
-
-        # A-side termination
-        termination_a = (
-            CircuitTermination.objects.restrict(request.user, "view")
-            .prefetch_related("site__region")
-            .filter(circuit=instance, term_side=CircuitTerminationSideChoices.SIDE_A)
-            .first()
-        )
-        if (
-            termination_a
-            and termination_a.connected_endpoint
-            and hasattr(termination_a.connected_endpoint, "ip_addresses")
-        ):
-            termination_a.ip_addresses = termination_a.connected_endpoint.ip_addresses.restrict(request.user, "view")
-
-        # Z-side termination
-        termination_z = (
-            CircuitTermination.objects.restrict(request.user, "view")
-            .prefetch_related("site__region")
-            .filter(circuit=instance, term_side=CircuitTerminationSideChoices.SIDE_Z)
-            .first()
-        )
-        if (
-            termination_z
-            and termination_z.connected_endpoint
-            and hasattr(termination_z.connected_endpoint, "ip_addresses")
-        ):
-            termination_z.ip_addresses = termination_z.connected_endpoint.ip_addresses.restrict(request.user, "view")
-
-        return {
-            "termination_a": termination_a,
-            "termination_z": termination_z,
-        }
-
-
-class CircuitEditView(generic.ObjectEditView):
-    queryset = Circuit.objects.all()
-    model_form = forms.CircuitForm
-    template_name = "circuits/circuit_edit.html"
-
-
-class CircuitDeleteView(generic.ObjectDeleteView):
+        context = super().get_extra_context(request, instance)
+        if self.action == "retrieve":
+            circuits = (
+                Circuit.objects.restrict(request.user, "view")
+                .filter(provider=instance)
+                .prefetch_related("type", "tenant", "terminations__site")
+            )
+            circuits_table = tables.CircuitTable(circuits)
+            circuits_table.columns.hide("provider")
+
+            paginate = {
+                "paginator_class": EnhancedPaginator,
+                "per_page": get_paginate_count(request),
+            }
+            RequestConfig(request, paginate).configure(circuits_table)
+
+            context["circuits_table"] = circuits_table
+        return context
+
+
+class CircuitUIViewSet(NautobotUIViewSet):
+    model = Circuit
+    bulk_create_form_class = forms.CircuitCSVForm
+    bulk_update_form_class = forms.CircuitBulkEditForm
+    filterset_class = filters.CircuitFilterSet
+    filterset_form_class = forms.CircuitFilterForm
+    form_class = forms.CircuitForm
+    lookup_field = "pk"
```

Comment:
The new Object*Views default to `slug` lookup as this is our planned direction, changing the base from using PK.